scalalib/test/src/mill/scalalib/HelloJavaTests.scala

```diff
@@ -47,6 +63,23 @@ object HelloJavaTests extends TestSuite {
         !os.walk(result3.value.classes.path).exists(_.last == "Core.class")
       )
     }
+
+    test("javaHome") {
+      val eval = UnitTester(HelloJavaJavaHomeOverride, resourcePath)
+
+      val Right(result) = eval.apply(HelloJavaJavaHomeOverride.core.compile)
+
+      val coreClassFile = os.walk(result.value.classes.path).find(_.last == "Core.class")
+
+      assert(
+        coreClassFile.isDefined,
+
+        // The first eight bytes are magic numbers followed by two bytes for major version and two bytes for minor version
+        // We are overriding to java 23 which corresponds to class file version 67
+        os.read.bytes(coreClassFile.get, 4, 4).toSeq == Seq[Byte](0, 0, 0, 67)
```

Comment:
Let's also add a test to exercise `.test` and `.run` and verify that `System.getProperty("java.version")` returns the expected value. We should also exercise this for two different java versions, to verify that it's not just picking up the environmental default

Reply:
Should overriding the java version also affect `.launcher` and `.assembly`? 

* `launcher` is explicitly not meant to be portable: it's meant to be something you run on the same machine where it was built, so any JVM you installed would still be present

* `assembly` is meant to be portable, e.g. run on a different machine from which it was built, so we probably can't rely on the configured java home for that